### PR TITLE
Migrate tests to roslyn test libraries

### PR DESCRIPTION
**Pull request by AlexanderSher**

### CODE REVIEW

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs [hunk 1]
```diff
@@ -2,75 +2,67 @@
...
-            Assert.Equal("AZC0001", diagnostic.Id);
-            Assert.Equal("Namespace 'RandomNamespace' shouldn't contain public types. Use one of the following pre-approved namespace groups: " +
-                         "Azure.AI, Azure.Analytics, Azure.Data, Azure.Iot, Azure.Media, Azure.Messaging, Azure.Security, Azure.Storage, Microsoft.Extensions.Azure", diagnostic.GetMessage());
+}";
 
-            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostic.Location);
+            var diagnostic = Verifier.Diagnostic("AZC0001")
+                .WithMessage("Namespace 'RandomNamespace' shouldn't contain public types. Use one of the following pre-approved namespace groups: " +
+                             "Azure.AI, Azure.Analytics, Azure.Data, Azure.Iot, Azure.Media, Azure.Messaging, Azure.Security, Azure.Storage, Microsoft.Extensions.Azure")
+                .WithSpan(2, 11, 2, 26);
```
Comment (pakrym) [inline src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs]: Why didn't you use '[||]` here?

**Comment by pakrym** [comment]
I've noticed you haven't deleted any of existing analyzer test framework files, are they still being used somewhere?

**Comment by AlexanderSher** [comment]
> I've noticed you haven't deleted any of existing analyzer test framework files, are they still being used somewhere?

Nope, just forgotten to delete them :)

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs (hunk 1 above)
Comment (AlexanderSher) [reply src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs]: I haven't found an easy way to explicitly assert the message. I can spend some time on digging, but I don't see much value in it.

**Pull request merged**